mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,709 @@
+        self.aug_ratio = _pair(aug_ratio)
+        if not mmcv.is_tuple_of(self.aug_ratio, (int, float)):
+            raise TypeError(f'aug_ratio should be int, float'
+                            f'or tuple of int and float, '
+                            f'but got {type(aug_ratio)}')
+        assert len(self.aug_ratio) == 2
+
+        total_ratio = (
+            self.sampler.positive_ratio + self.sampler.background_ratio +
+            self.sampler.incomplete_ratio)
+        self.positive_per_video = int(
+            self.sampler.num_per_video *
+            (self.sampler.positive_ratio / total_ratio))
+        self.background_per_video = int(
+            self.sampler.num_per_video *
+            (self.sampler.background_ratio / total_ratio))
+        self.incomplete_per_video = (
+            self.sampler.num_per_video - self.positive_per_video -
+            self.background_per_video)
+
+        self.test_interval = self.test_cfg.ssn.sampler.test_interval
+        # number of consecutive frames
+        self.clip_len = clip_len
+        # number of steps (sparse sampling for efficiency of io)
+        self.frame_interval = frame_interval
+
+        # test mode or not
+        self.filter_gt = filter_gt
+        self.test_mode = test_mode
+
+        # yapf: disable
+        if self.verbose:
+            logger = get_root_logger()
+            logger.info(f"""
+            SSNDataset: proposal file {self.proposal_file} parsed.
+
+            There are {len(self.positive_pool) + len(self.background_pool) +
+                len(self.incomplete_pool)} usable proposals from {len(self.video_infos)} videos.
+            {len(self.positive_pool)} positive proposals
+            {len(self.incomplete_pool)} incomplete proposals
+            {len(self.background_pool)} background proposals
+
+            Sample config:
+            FG/BG/INCOMP: {self.positive_per_video}/{self.background_per_video}/{self.incomplete_per_video}  # noqa:E501
+            Video Centric: {self.video_centric}
+
+            Regression Stats:
+            Location: mean {self.reg_stats[0][0]:.05f} std {self.reg_stats[1][0]:.05f}
+            Duration: mean {self.reg_stats[0][1]:.05f} std {self.reg_stats[1][1]:.05f}
+            """)
+        # yapf: enable
+        else:
+            print(f'SSNDataset: proposal file {self.proposal_file} parsed.')
+
+    def load_annotations(self):
+        """Load annotation file to get video information."""
+        video_infos = []
+        if 'normalized_' in self.ann_file:
+            self.proposal_file = self.ann_file.replace('normalized_', '')
+            if not osp.exists(self.proposal_file):
+                logger = get_root_logger()
+                logger.info(f'{self.proposal_file} does not exist.'
+                            f'Converting from {self.ann_file}')
+                frame_dict = parse_frame_folder(
+                    self.data_prefix, key_func=lambda x: x.split('/')[-1])
```

Comment:
`key_func=lambda: x: osp.basename(x)`

Reply:
Can we solve this by use annotation file that contain the `directory` and `num_frames`?